src/Catalog/AggregateCursor.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog
+{
+    /// <summary>
+    /// A <see cref="ReadCursor"/> that returns the greatest value for <see cref="Load(CancellationToken)"/> from a set of <see cref="ReadCursor"/>s based on an <see cref="IComparer{DateTime}"/>.
+    /// </summary>
+    public class AggregateCursor : ReadCursor
+    {
+        public class Least : IComparer<DateTime>
+        {
+            public int Compare(DateTime x, DateTime y)
+            {
+                return Math.Sign((y- x).Ticks);
+            }
+        }
+
+        public AggregateCursor(IEnumerable<ReadCursor> innerCursors)
+            : this(innerCursors, new Least())
+        {
+        }
+
+        public AggregateCursor(IEnumerable<ReadCursor> innerCursors, IComparer<DateTime> comparer)
```

Comment:
I was trying to think about the possible scenario that somebody wants to use this class for aggregating cursors with whatever `IComparer<DateTime>` they want

in retrospect, this doesn't seem very worthwhile, as someone would likely always want to start from the lowest cursor even if they're using this as the min, so i'll remove the comparer stuff